File: python-modular/testcases/preproc.py
"""
Test PreProc
"""

from shogun.Kernel import *

import util

########################################################################
# kernel computation
########################################################################

def _kernel (indata, feats):
	fun=eval(indata['name_kernel']+'Kernel')
	args=util.get_args(indata, 'kernel_arg')

	kernel=fun(feats['train'], feats['train'], *args)
	ktrain=max(abs(indata['km_train']-kernel.get_kernel_matrix()).flat)
	kernel.init(feats['train'], feats['test'])
	ktest=max(abs(indata['km_test']-kernel.get_kernel_matrix()).flat)

	return util.check_accuracy(indata['accuracy'], ktrain=ktrain, ktest=ktest)

########################################################################
# public
########################################################################

def test (indata):
	fun=eval('util.get_feats_'+indata['feature_class'])
	feats=fun(indata)
	feats=util.add_preproc(indata['name'], feats)
	return _kernel(indata, feats)

File: testsuite/octave-modular/preproc.m
function y = preproc(filename)
	init_shogun;
	y=true;

	addpath('util');
	addpath('../data/preproc');

	eval('globals'); % ugly hack to have vars from filename as globals
	eval(filename);

	if strcmp(name, 'LogPlusOne')==1
		preproc=LogPlusOne();
	elseif strcmp(name, 'NormOne')==1
		preproc=NormOne();
	elseif strcmp(name, 'PruneVarSubMean')==1
		preproc=PruneVarSubMean(tobool(preproc_arg0_divide));
	elseif strcmp(name, 'SortUlongString')==1
		preproc=SortUlongString();
	elseif strcmp(name, 'SortWord')==1
		preproc=SortWord();
	elseif strcmp(name, 'SortWordString')==1
		preproc=SortWordString();
	else
		error('Unsupported preproc %s', name);
	end

	if ~set_features()
		return;
	end

	preproc.init(feats_train);
	feats_train.add_preproc(preproc);
	feats_train.apply_preproc();
	feats_test.add_preproc(preproc);
	feats_test.apply_preproc();

	if ~set_kernel()
		return;
	end

	ktrain=max(max(abs(km_train-kernel.get_kernel_matrix())));
	kernel.init(feats_train, feats_test);
	ktest=max(max(abs(km_test-kernel.get_kernel_matrix())));

	data={'kernel', ktrain, ktest};
	y=check_accuracy(accuracy, data);
